File: examples/class-kits/gospider/api_bootstrap.go
package classkit

import scrapyapi "gospider/scrapy"

func NewAPIBootstrapSpider() *scrapyapi.Spider {
	return scrapyapi.NewSpider("api-bootstrap", func(response *scrapyapi.Response) ([]any, error) {
		return []any{
			scrapyapi.NewItem().
				Set("kind", "api_bootstrap").
				Set("title", response.CSS("title").Get()).
				Set("url", response.URL).
				Set("bootstrap_excerpt", response.Text),
		}, nil
	}).AddStartURL("https://example.com/app/page")
}

File: examples/class-kits/rustspider/api_bootstrap.rs
use std::sync::Arc;

use rustspider::scrapy::project as projectruntime;
use rustspider::scrapy::{Item, Output, Response, Spider};

pub fn make_api_bootstrap_spider() -> Spider {
    Spider::new(
        "api-bootstrap",
        Arc::new(|response: &Response| {
            vec![Output::Item(
                Item::new()
                    .set("kind", "api_bootstrap")
                    .set("title", response.css("title").get().unwrap_or_default())
                    .set("url", response.url.clone())
                    .set("bootstrap_excerpt", response.text.clone()),
            )]
        }),
    )
    .add_start_url("https://example.com/app/page")
}

pub fn register() {
    projectruntime::register_spider("api-bootstrap", make_api_bootstrap_spider);
}

